### ISSUE TITLE
UI is out of sync with selection in Send Finalized Form after submission failure

### ISSUE THREAD

**Issue by lognaturel** [getodk/collect#3866]
#### Software and hardware versions 
Collect v1.27.0

#### Problem description
UI is out of sync with selection in Send Finalized Form after submission failure.

#### Steps to reproduce the problem
1. Select a form that will fail to send
1. Send it 
1. See that it is still selected but if you try to send again, you get an error saying nothing is selected

#### Expected behavior
The UI should update to match the selection state.

**Comment by mmarciniak90** [comment]
@lognaturel is it possible that this issue is related to the Android version? Or maybe some extra conditions need to be met. Maybe the cause of failure is important.
I'm not able to reproduce this on Android: 6.0, 7.0, 8.1, 9.0
I try to send form which is not available on Aggregate.

**Comment by lognaturel** [comment]
Huh! That is really strange and I guess it explains why you've never run into it. 🙃 I'm on Android 7.0 but can't really think of how it would be related to Android version. I'll try to get a reproduction on a sandbox server. If you haven't been able to reproduce it's not very important.

**Comment by mmarciniak90** [comment]
Yeah! Finally, we were able to reproduce this issue. It seems that different causes of failure give different results on Android versions. Here are some examples.

CASE 1 : 
<img height="40%" width="40%" src="https://user-images.githubusercontent.com/3907554/112136082-cc2da780-8bce-11eb-93ee-12ab9d5f6a0d.png">

1. Form downloaded from the default demo server
2. I edited the server URL
3. I sent form -> issue wasn't visible, I tried sent again -> **issue visible on Android 7, 8.1, 10**
4. I reopened the `Send Finalized Form` view and tried again -> issue visible only on Android 10, second try also caused it on Android 7 and 8

CASE 2:
<img height="40%" width="40%" src="https://user-images.githubusercontent.com/3907554/112136788-b967a280-8bcf-11eb-8ce8-9e41f11a2d12.png">

1. Form downloaded from the default demo server
2. I edited the server URL to Aggregate 
3. I sent form -> issue wasn't visible, I tried sent again -> **issue was visible only on Android 7**
4. I reopened the `Send Finalized Form` view and tried again -> issue was visible only on Android 7

CASE 3:

<img height="40%" width="40%" src="https://user-images.githubusercontent.com/3907554/112137608-bfaa4e80-8bd0-11eb-9a38-020f1682c9f2.png">

1. Form downloaded from the default demo server
2. I scanned the Central QR code where this form was not available
3. I sent form -> issue was visible only on Android 10 -> I tried again on Android 7 and 8 and then the issue was visible

**Comment by kkrawczyk123** [comment]
I was able to reproduce the issue on all my Android devices, there are my results:

- When I input in the URL my local aggregate with the wrong port number. On Android 9 it was visible in my first try but on 7.1 and 5.1 in the second try (after the first one form was not selected after failed sending).
the error message that I have seen:
![Screenshot_20210323-123916](https://user-images.githubusercontent.com/33342488/112142698-364a4a80-8bd7-11eb-8a6d-b3c10b5fcfbf.png)

- I was also able to reproduce it with the wrong URL of Central when input random letter in URL but not in // with project number. Here I was able to reproduce it only in the second try on all devices. 
the error message that I have seen:
![Screenshot_20210323-124026](https://user-images.githubusercontent.com/33342488/112143320-02235980-8bd8-11eb-86b2-3c4525d85178.png)


- I was also able to reproduce when I downloaded form from Central demo server and tried to send a submission to aggregate where this form does not exist. On Android 9 in the first try but on 7.1 and 5.1 in the second try. I obtained the same error message that @mmarciniak90 in Case 2.

One more interesting fact is that form selection after failed sending disappears after screen rotation.
